agent.go

```diff
@@ -385,6 +385,100 @@ func (c *controller) agentClose() {
 	agent.networkDB.Close()
 }
 
+type Task struct {
+	Name       string
+	EndpointID string
+	EndpointIP string
+	Info       string
+}
+
+type ServiceInfo struct {
+	VIP   string
+	Tasks []Task
+	Ports []string
+}
+type epRecord struct {
+	ep   EndpointRecord
+	info string
+}
+
+func (n *network) Services() map[string]ServiceInfo {
+	eps := make(map[string]epRecord)
+
+	if !n.isClusterEligible() {
+		return nil
+	}
+	agent := n.getController().getAgent()
+	if agent == nil {
+		return nil
+	}
+
+	// Walk through the libnetwork's "endpoint_table" and fetch the driver
+	// agnostic endpoint info
+	entries := agent.networkDB.GetTableByNetwork("endpoint_table", n.id)
+	for eid, value := range entries {
+		var epRec EndpointRecord
+		if err := proto.Unmarshal(value.([]byte), &epRec); err != nil {
+			logrus.Errorf("Unmarshal of %s failed", eid)
+			continue
+		}
+		eps[eid] = epRecord{
+			ep: epRec,
+		}
+	}
+
+	// Walk through the driver's tables, have the driver decode the entries
+	// and return the tuple {ep ID, value}. value is a string that coveys
+	// relevant info about the endpoint.
+	d, err := n.driver(true)
+	if err != nil {
+		logrus.Errorf("Can't get the driver for network, %s\n", n.ID())
+		return nil
+	}
+	for _, table := range n.driverTables {
+		entries := agent.networkDB.GetTableByNetwork(table, n.id)
+		for key, value := range entries {
+			epID, info := d.DecodeTableEntry(table, key, value.([]byte))
+			if ep, ok := eps[epID]; !ok {
+				logrus.Errorf("epRecord update attempt for driver info, but missing for %s", epID)
```

Comment:
Please reword.
This could be a real discrepancy between driver state and libnetwork core state, but it could also be a genuine difference if a new task was scheduled on this node while running the inspect verbose command.

IMO, given this is for diagnostic, we should also report the spurious driver enpoint entry and mark it as such.
We could have an extra flag in `epRecord` like as an example `Inconsistent` which the network inspect cli will print only if true.